code-challenges.js

```diff
@@ -51,8 +112,39 @@ var vowelTester2 = "throw"
 var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
 
 
+//create a function that take in an array
+const catsOnly = (array) => {
+    //use filter to filter through array
+    return array.filter(value => {
+        // find only where animal key is equal cat
+        if (value.animal == "cat") {
+            // return that value (or object) to the filtered array
+            return value
+        }
+    })
+}
+
+console.log(catsOnly(toonimals))
 
 
 
 // --------------------6) Using the toonimals variable, create a function that returns only the names of the non-cats.
 // Expected output: "Itchy" "Daffy" "Ren"
+
+//create a function that take in an array
+const nonCatsOnly = (array) => {
+    // create a new array to store names only
+    let namesOnly = []
+    //use filter to filter through array
+    array.filter(value => {
+        // find only where animal key is not equal to cat
+        if (value.animal !== "cat") {
+            // return the name only of that value (or object) to the new array
+            return namesOnly.push(value.name)
+        }
+    })
+
+    return namesOnly.join(" ")
+}
+
+console.log(nonCatsOnly(toonimals))
```

Comment:
YES! Excellent work taking in to account the fact the expected output was not in an array! 🥇